VideoWeb/VideoWeb/ClientApp/src/app/shared/toast/room-closing/room-closing-toast.component.ts

```diff
@@ -0,0 +1,55 @@
+import * as moment from 'moment';
+import { Component, OnInit } from '@angular/core';
+import { ToastrService, ToastPackage } from 'ngx-toastr';
+import { Observable } from 'rxjs';
+import { VhToastButton, VhToastComponent } from '../vh-toast.component';
+import { ClockService } from 'src/app/services/clock.service';
+import { map, startWith } from 'rxjs/operators';
+
+interface RoomClosingToastOptions {
+    expiryDate: Date;
+    buttons: VhToastButton[];
+}
+@Component({
+    templateUrl: './room-closing-toast.component.html',
+    styleUrls: ['../vh-toast.component.scss']
+})
+export class RoomClosingToastComponent extends VhToastComponent implements OnInit {
+    alertMessage$: Observable<string>;
+    msAllowedForPrivateConsultationsAfterClosing: number;
+    expiryDate: Date;
+
+    set roomClosingToastOptions(options: RoomClosingToastOptions) {
+        this.expiryDate = options.expiryDate;
+        super.vhToastOptions = {
+            color: 'white',
+            onNoAction: async () => {
+                // this intentionally does nothing
+            },
+            buttons: options.buttons
+        };
+    }
+
+    constructor(protected toastrService: ToastrService, public toastPackage: ToastPackage, private clockService: ClockService) {
+        super(toastrService, toastPackage);
+    }
+
+    ngOnInit(): void {
+        this.alertMessage$ = this.clockService.getClock().pipe(
+            startWith(new Date()),
+            map(date => {
+                return this.calcTimeLeft(date);
+            })
+        );
```

Comment:
does this need a `.subscribe()` ?

Reply:
no because it's being subscribed to by the `| async` pipe in the UI - you only need to use subscribe here if
* you're doing something with a value emitted
* you're not using the async pipe in the UI (this handles the subscription & unsubscription for you)

Reply:
cool